### DISCUSSION TITLE
How can I use _simplefix_ for FIX regression testing?

### DISCUSSION THREAD

**Discussion by da4089** [da4089/simplefix#55] [Q&A]
Migrated from  #43, where @mecanda12 asked:

> I will be using this library for testing purpose only for regression. I do not think need history and database. All I need establish the Network, that is where I need help.
> My understanding is : Regression Test suite -----> Simplefix library-------> Trading engine
> 
> By the way, Do I need to set up a fix server separately to connect with trading engine?

Ok, that's something that _simplefix_ can do well, especially when it comes to testing corner-cases of malformed messages, recovery scenarios, etc.  You **do not** need a separate FIX engine for this.

I will dig up some example code and post it.

**Comment by mecanda12** [comment]
@da4089 Thanks for digging for an example of this.

**Comment by mecanda12** [comment]
@da4089 @gjvc @JonathonMurphy [jvirtanen](https://github.com/jvirtanen)  Any update on this?

**Comment by mecanda12** [comment]
@da4089 Do you think the code will work with this library?

import socket
import simplefix

def establish_fix_connection(host, port):
    # Create a socket object
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

    try:
        # Connect to the FIX server
        s.connect((host, port))
        print(f"Connected to {host}:{port}")
        
        # Send a Logon message to the server
        msg = simplefix.FixMessage()
        msg.append_pair(35, "A")
        msg.append_pair(49, "SenderCompID")
        msg.append_pair(56, "TargetCompID")
        msg.append_pair(34, "1")
        msg.append_pair(52, "20210101-00:00:00.000")
        msg.append_pair(98, "0")
        msg.append_pair(108, "30")
        msg.append_pair(141, "Y")
        msg.append_pair(553, "Username")
        msg.append_pair(554, "Password")
        s.send(msg.encode())

        # Receive a response from the server
        response = s.recv(1024)
        print(f"Response received: {response}")

    except Exception as e:
        print(f"Error establishing FIX connection: {e}")

    # Close the socket connection
    s.close()

**Reply by da4089** [reply]
That's more or less how I'd expect it to be used, yes.  I've written similar code myself.

You might want to look at the `FixMessage.append_utc_timestamp()` function (or one of the other similar ones), which can make adding timestamps easier.

**Reply by mecanda12** [reply]
@da4089 Thanks for the update on the Timestamp function. Is there any feedback? By the way, this code still does not work. Can you help with this, please?
